packages/serve/src/lib/middleware/response-format/helpers.ts

```diff
@@ -28,23 +13,37 @@ export const isReceivedFormatRequest = (
  */
 export const checkUsableFormat = ({
   context,
-  formatters,
   supportedFormats,
   defaultFormat,
 }: {
   context: KoaContext;
-  formatters: ResponseFormatterMap;
   supportedFormats: string[];
   defaultFormat: string;
 }) => {
-  for (const format of supportedFormats) {
-    if (!(format in formatters)) continue;
-    if (!isReceivedFormatRequest(context, format)) continue;
-    return format;
+  // find last matched value be format
+  const pathFormat = context.path.split('.')[1];
+
+  // match result for searching in Accept header.
+  const acceptFormat = context.accepts(supportedFormats);
+
+  // if path ending has no format
+  if (!pathFormat) {
+    // get default when "Accept" header also not matched or "Accept" header not in request (shows by */*)
+    if (!acceptFormat || acceptFormat == '*/*') return defaultFormat;
```

Comment:
The case `*/*` happened when I use the `supertest`, because `supertest` won't provide `Accept`  ( `Accept` is undefined ).

The logic from the `koa.accepts` will return `*/*`. The koa use the [npm accepts](https://www.npmjs.com/package/accepts) and [it show the case in test case](https://github.com/jshttp/accepts/blob/master/test/type.js#L17).

If we trace the code, `accepts` use the the [npm negotiator](https://www.npmjs.com/package/negotiator) be the core library. And the negotiator will return `*/*` when [Accept value is undefined](https://github.com/jshttp/negotiator/blob/master/test/mediaType.js#L8)